toolchain/check/convert.cpp

```diff
@@ -1082,6 +1116,19 @@ static auto PerformBuiltinConversion(Context& context, SemIR::LocId loc_id,
              .type_inst_id = type_inst_id,
              .witnesses_block_id = lookup_result.inst_block_id()});
       }
+    } else {
+      // If impl lookup fails, don't keep looking for another way to convert.
+      // See https://github.com/carbon-language/carbon-lang/issues/5122.
+      // TODO: Move into `LookupImplWitness` so it can add notes explaining
```

Comment:
It could also perhaps pass in this function as a lambda to construct the error, and the lookup could add notes to that.

Reply:
Yeah, that is how it probably will work. I have updated the TODO.